consensus.go

```diff
@@ -0,0 +1,181 @@
+package dcrlibwallet
+
+import (
+	"fmt"
+	"sort"
+
+	w "decred.org/dcrwallet/v2/wallet"
+	"github.com/decred/dcrd/chaincfg/chainhash"
+	"github.com/decred/dcrd/chaincfg/v3"
+)
+
+// SetVoteChoice sets a voting choice for the specified agenda. If a ticket
+// hash is provided, the voting choice is also updated with the VSP controlling
+// the ticket. If a ticket hash isn't provided, the vote choice is set for all
+// tickets controlled by the specified VSP.
+func (wallet *Wallet) SetVoteChoice(vspHost string, vspPubKey []byte, agendaID, choiceID, hash string, passphrase []byte) error {
+	wal := wallet.Internal()
+
```

Comment:
Not sure this is needed anymore. Please just use `wallet.Internal()` directly instead. It's a little confusing seeing usages of `wal.***` and `wallet.***`.
```suggestion
```